miio/integrations/vacuum/roborock/vacuumcontainers.py

```diff
@@ -163,33 +184,76 @@ def error(self) -> str:
             return "Definition missing for error %s" % self.error_code
 
     @property
-    @sensor("Battery", unit="%", device_class="battery")
+    @sensor(
+        "Dock error code",
+        icon="mdi:alert",
+        entity_category="diagnostic",
+        enabled_default=False,
+    )
+    def dock_error_code(self) -> int:
+        """Dock error status as returned by the device."""
+        return int(self.data["dock_error_status"])
+
+    @property
+    @sensor(
+        "Dock error string",
+        icon="mdi:alert",
+        entity_category="diagnostic",
+        enabled_default=False,
+    )
+    def dock_error(self) -> str:
+        """Human readable dock error description, see also :func:`dock_error_code`."""
+        try:
+            return dock_error_codes[self.dock_error_code]
+        except KeyError:
+            return "Definition missing for dock error %s" % self.dock_error_code
+
+    @property
+    @sensor("Battery", unit="%", device_class="battery", enabled_default=False)
     def battery(self) -> int:
         """Remaining battery in percentage."""
         return int(self.data["battery"])
 
     @property
-    @setting(
-        "Fanspeed",
-        unit="%",
-        setter_name="set_fan_speed",
-        min_value=0,
-        max_value=100,
-        step=1,
-        icon="mdi:fan",
-    )
```

Comment:
I removed this because it does not work with my Roborock S7 MaxV+.
That only excepts fan_speed values of 101 to 108 (the same settings as in the app), not 0 to 100.
Those fan speeds are already exposed through the vacuum entity in HomeAssistant.

We could add this back for other models that might support it, but we schould then make some model decorator or something to specifiy which models support this.
I would want to leave that to a future PR.

Reply:
I think we should probably keep this but return `None` if the value > 100, assuming this could be used as a proxy to check if the device supports defining the fan speed in percentage?

Reply:
Sounds good, I will do that.

Reply:
Done